### ISSUE TITLE
Local variable is not used

### ISSUE THREAD

**Issue by romanorac** [PyCQA/pylint#1002]
```
def some_fun():
    some_string = 'some_string'
    var1 = some_string.replace('some', 'that')
    var1 = some_string.replace('some', 'that')
    return var1
```

Currently, pylint reports nothing, but it should report a warning: Local variable var1 is not used.

pylint 1.6.1,
astroid 1.4.7
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  5 2015, 21:12:44)
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]

**Comment by PCManticore** [comment]
Hi, thanks for the report.

Is this the full code? Since I don't have any error regarding var1, here's my output:

```
    (lala) D:\Projects\repos\tests>python -m pylint a.py -rn
    No config file found, using default configuration
    ************ Module a
    :  5, 0: Final newline missing (missing-final-newline)
    :  1, 0: Missing module docstring (missing-docstring)
    :  1, 0: Missing function docstring (missing-docstring)
```

Can you paste the full snippet for which this reproduces?

**Comment by romanorac** [comment]
I’m sorry I wasn’t more clear. You output is ok, but there should be an additional warning.
```
def some_fun():
    some_string = 'some_string'
    var1 = some_string.replace('some', 'that') # here we declare a variable var1
    var1 = some_string.replace('some', 'that') # here reassign a value to var1 and shadow previous declaration
    return var1
```
PyCharm shows greyed variable as seen on the picture bellow:
![screen shot 2016-07-14 at 11 46 57](https://cloud.githubusercontent.com/assets/8342728/16835370/ba26f81e-49b8-11e6-8d7f-c0c0b90813ec.png)

**Comment by PCManticore** [comment]
Not sure this is actually worth having, as it's a prevalent pattern that's going to make red all the CIs running pylint.
